diff --git a/gtk3.spec b/gtk3.spec
index 62e0d4f..0d7d630 100644
--- a/gtk3.spec
+++ b/gtk3.spec
@@ -2,6 +2,7 @@
 %global with_broadway 1
 %endif
 
+%global ubuntu_release 3ubuntu2
 %global glib2_version 2.57.2
 %global pango_version 1.41.0
 %global atk_version 2.35.1
@@ -17,7 +18,7 @@
 # Filter provides for private modules
 %global __provides_exclude_from ^%{_libdir}/gtk-3.0
 
-Name:    gtk3
+Name:    gtk3-ubuntu
 Version: 3.24.34
 Release: 2%{?dist}
 Summary: GTK+ graphical user interface library
@@ -25,7 +26,11 @@ Summary: GTK+ graphical user interface library
 License: LGPLv2+
 URL:     https://gtk.org
 Source0: https://download.gnome.org/sources/gtk+/3.24/gtk+-%{version}.tar.xz
+Source1: http://archive.ubuntu.com/ubuntu/pool/main/g/gtk+3.0/gtk+3.0_%{version}-%{ubuntu_release}.debian.tar.xz
+Patch0:  add-files-to-meson.patch
 
+Obsoletes:     gtk3
+Provides:      gtk3%{?_isa} = %{version}
 BuildRequires: pkgconfig(atk) >= %{atk_version}
 BuildRequires: pkgconfig(atk-bridge-2.0)
 BuildRequires: pkgconfig(avahi-gobject)
@@ -99,6 +104,8 @@ This package contains version 3 of GTK+.
 
 %package -n gtk-update-icon-cache
 Summary: Icon theme caching utility
+Obsoletes: gtk3-gtk-update-icon-cache
+Provides:  gtk3-gtk-update-icon-cache%{?_isa} = %{version}
 
 %description -n gtk-update-icon-cache
 GTK+ can use the cache files created by gtk-update-icon-cache to avoid a lot of
@@ -108,6 +115,8 @@ applications, the overall memory consumption is reduced as well.
 
 %package immodules
 Summary: Input methods for GTK+
+Obsoletes: gtk3-immodules
+Provides:  gtk3-immodules%{?_isa} = %{version}
 Requires: gtk3%{?_isa} = %{version}-%{release}
 # for im-cedilla.conf
 Requires: gtk2-immodules%{?_isa}
@@ -118,6 +127,8 @@ are shipped as part of GTK+ 3.
 
 %package immodule-xim
 Summary: XIM support for GTK+
+Obsoletes: gtk3-immodules-xim
+Provides:  gtk3-immodules-xim%{?_isa} = %{version}
 Requires: gtk3%{?_isa} = %{version}-%{release}
 
 %description immodule-xim
@@ -125,6 +136,8 @@ The gtk3-immodule-xim package contains XIM support for GTK+ 3.
 
 %package devel
 Summary: Development files for GTK+
+Obsoletes: gtk3-devel
+Provides:  gtk3-devel%{?_isa} = %{version}
 Requires: gtk3%{?_isa} = %{version}-%{release}
 
 %description devel
@@ -135,6 +148,8 @@ gtk3-devel-docs package.
 
 %package devel-docs
 Summary: Developer documentation for GTK+
+Obsoletes: gtk3-devel-docs
+Provides:  gtk3-devel-docs%{?_isa} = %{version}
 Requires: gtk3 = %{version}-%{release}
 
 %description devel-docs
@@ -143,6 +158,8 @@ widget toolkit.
 
 %package tests
 Summary: Tests for the %{name} package
+Obsoletes: gtk3-tests
+Provides:  gtk3-tests%{?_isa} = %{version}
 Requires: %{name}%{?_isa} = %{version}-%{release}
 
 %description tests
@@ -150,7 +167,14 @@ The %{name}-tests package contains tests that can be used to verify
 the functionality of the installed %{name} package.
 
 %prep
-%autosetup -n gtk+-%{version} -p1
+%autosetup -n gtk+-%{version} -N -p1
+
+tar Jxvf "%{SOURCE1}"
+
+for i in $(grep -v '#' debian/patches/series); do
+  patch -Np1 -i "debian/patches/${i}"
+done
+patch -u -b gtk/meson.build -i "%{PATCH0}"
 
 %build
 export CFLAGS='-fno-strict-aliasing %optflags'
